cloudcredential/v1/types_aws.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2018 The OpenShift Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TODO: these types should eventually be broken out, along with the actuator, to a separate repo.
+
+// AWSProviderSpec contains the required information to create a user policy in AWS.
+// +k8s:deepcopy-gen=false
+type AWSProviderSpec struct {
+	metav1.TypeMeta `json:",inline"`
+	// StatementEntries contains a list of policy statements that should be associated with this credentials access key.
+	StatementEntries []StatementEntry `json:"statementEntries"`
+}
+
+// StatementEntry models an AWS policy statement entry.
+// +k8s:deepcopy-gen=false
+type StatementEntry struct {
+	// Effect indicates if this policy statement is to Allow or Deny.
+	Effect string `json:"effect"`
```

Comment:
this is an enum? Add kubebuilder markers.

Reply:
I believe today the only allowed values are "Allow" or "Deny" so I supposes we can make this an enum.

Reply:
Same question here, for an embedded type behind a RawExtension, what effect does annotating these fields have? (I annotated it as an enum, and there was no difference in the generated CRD which confused me until I realized this was behind the spec.providerSpec RawExtension.)

Reply:
That's a kubebuilder question. It might have special code to generate a RawExtensins schema.